internal/users/users.go

```diff
@@ -14,17 +14,46 @@ import (
 	"github.com/gorilla/mux"
 	"github.com/lib/pq"
 	"golang.org/x/crypto/bcrypt"
+	"gopkg.in/go-playground/validator.v9"
 	"net/http"
 	"net/smtp"
 	"net/url"
+	"strings"
 	"time"
 )
 
+type UserValid struct {
+	validate          *validator.Validate
+	failedValidations []string
+	validErr          map[string]string
```

Comment:
having two of these seems excessive, I think one should be enough. Or what's the difference between the two?